Bottom: 7b9294b0d7eb1ce49ed1ce396d1a050611305b8f
Top:    bf784c2375014c3c563ffa08e8e1f2acdee32514
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2008-11-01 00:45:26 -0400

Fix ups to make mmc over spi work










---

diff --git a/arch/powerpc/boot/dts/dspeak01.dts b/arch/powerpc/boot/dts/dspeak01.dts
index 35872b3..ec163c0 100644
--- a/arch/powerpc/boot/dts/dspeak01.dts
+++ b/arch/powerpc/boot/dts/dspeak01.dts
@@ -176,9 +176,9 @@
 			interrupt-parent = <&mpc5200_pic>;
 
 			mmc-slot@0 {
-				compatible = "mmc-spi";
+				compatible = "linux,mmc-spi";
 				reg = <0>;
-				max-speed = <50000000>;
+				spi-max-frequency = <20000000>;
 				/* Unregulated slot. */
 				voltage-range = <3300 3300>;
 				/*gpios = <&sdcsr_pio 1 0   /*  WP */
diff --git a/drivers/mmc/host/mmc_spi.c b/drivers/mmc/host/mmc_spi.c
index 07faf54..b1d27bb 100644
--- a/drivers/mmc/host/mmc_spi.c
+++ b/drivers/mmc/host/mmc_spi.c
@@ -1410,7 +1410,7 @@ static int __devexit mmc_spi_remove(struct spi_device *spi)
 
 static struct spi_driver mmc_spi_driver = {
 	.driver = {
-		.name =		"mmc_spi",
+		.name =		"mmc-spi",
 		.bus =		&spi_bus_type,
 		.owner =	THIS_MODULE,
 	},
diff --git a/drivers/of/of_mmc_spi.c b/drivers/of/of_mmc_spi.c
index aa4e017..f189cdb 100644
--- a/drivers/of/of_mmc_spi.c
+++ b/drivers/of/of_mmc_spi.c
@@ -95,7 +95,7 @@ static int of_mmc_spi_add(struct device *dev)
 	const u32 *voltage_range;
 	int size;
 
-	if (!np || !of_device_is_compatible(np, "mmc-spi"))
+	if (!np || !of_device_is_compatible(np, "linux,mmc-spi"))
 		return NOTIFY_DONE;
 
 	oms = kzalloc(sizeof(*oms), GFP_KERNEL);
@@ -152,7 +152,7 @@ static int of_mmc_spi_del(struct device *dev)
 	struct device_node *np = dev->archdata.of_node;
 	struct of_mmc_spi *oms;
 
-	if (!np || !of_device_is_compatible(np, "mmc-spi") ||
+	if (!np || !of_device_is_compatible(np, "linux,mmc-spi") ||
 			!dev->platform_data)
 		return NOTIFY_DONE;
